draft-js-mention-plugin/src/MentionSuggestions/index.js

```diff
@@ -10,35 +10,25 @@ import defaultEntryComponent from './Entry/defaultEntryComponent';
 
 export class MentionSuggestions extends Component {
   static propTypes = {
+    open: PropTypes.bool.isRequired,
+    onOpenChange: PropTypes.func.isRequired,
     entityMutability: PropTypes.oneOf(['SEGMENTED', 'IMMUTABLE', 'MUTABLE']),
     entryComponent: PropTypes.func,
     onAddMention: PropTypes.func,
-    suggestions: PropTypes.array,
+    suggestions: PropTypes.array.isRequired,
   };
 
   state = {
-    isActive: false,
     focusedOptionIndex: 0,
   };
 
-  UNSAFE_componentWillMount() {
+  constructor(props) {
+    super(props);
     this.key = genKey();
     this.props.callbacks.onChange = this.onEditorStateChange;
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (nextProps.suggestions.length === 0 && this.state.isActive) {
-      this.closeDropdown();
-    } else if (
-      nextProps.suggestions.length > 0 &&
-      nextProps.suggestions !== this.props.suggestions &&
-      !this.state.isActive
-    ) {
-      this.openDropdown();
-    }
-  }
```

Comment:
what do we do now to replace this behaviour?

Reply:
Allow user to handle it. Most of the time it's not desired behaviour. It forces to have suggestions in local state instead of being inferred which leads to complications when we need to update this state from props.